docs/DeveloperGuide.md

```diff
@@ -1,38 +1,177 @@
 # Developer Guide
 
-## Acknowledgements
+## Design
```

Comment:
It would be better to add overview list of the DG so that user can easily find the contents they want to refer.